planet/src/main/java/com/google/planet/data/ItineraryGenerator.java

```diff
@@ -32,31 +40,59 @@
         int END = events.get(0).getOpeningHours().end();
 
         Collections.sort(events, Event.SortByOrder);
-        int [][] travelTimeGraph = getTravelTimeGraph(events);
+        int [][] travelTimeGraph = getEventsDistanceMatrix(events);
         List<ItineraryItem> items = scheduleItineraryInOrder(events, START, END, travelTimeGraph);
         return items;
     }
 
     // Function that creates a graph with the events as vertices and the travelling
     // time between the events as edges.
-    // For the prototype, all travelling times are set to 15 minutes 
-    public int[][] getTravelTimeGraph(List<Event> events) {
+    // For the MVP, real time traffic is NOT used.
+    // TODO: handle invalid addresses
+    private int[][] getEventsDistanceMatrix(List<Event> events) {
         int numOfNodes = events.size();
         int[][] travelTimeGraph = new int[numOfNodes][numOfNodes];
```

Comment:
Add a comment explaining the structure of this data structure. What does each index represent?

Also, use a valuetype as the contained value: https://testing.googleblog.com/2017/11/obsessed-with-primitives.html

Reply:
Are you suggesting that I should define an object for the graph (I'm not really familiar with valuetypes)?

Reply:
As I see it, you have a few options. You could keep the graph as an array of arrays, but instead of having it contain an int, have it contain a Distance type. Distance should be a ValueType, which is basically a class which is immutable and has comparison operators.

Another option is to define a class for the entire graph, and have a function that returns the difference between two points.

int DistanceGraph.getDistance(address1, address2)

Reply:
Fixed